File: frontend/src/types/notification.ts
export enum NotificationType {
  EMAIL = 'EMAIL',
  SMS = 'SMS'
}

export enum NotificationStatus {
  PENDING = 'PENDING',
  PROCESSING = 'PROCESSING',
  COMPLETED = 'COMPLETED',
  FAILED = 'FAILED'
}

export interface NotificationCreate {
  notification_type: NotificationType;
  subject?: string;  // Optional - required for EMAIL, not used for SMS
  body: string;
  customer_ids: string[];
}

export interface Notification {
  id: string;
  notification_type: NotificationType;
  subject?: string;
  body: string;
  customer_ids: string[];
  status: NotificationStatus;
  created_at: string;
  updated_at: string;
}

File: notification-service/app/models/notification.py
from sqlalchemy import Column, String, DateTime, Enum, JSON, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime, timezone
import uuid
import enum

from app.database import Base


class NotificationType(enum.Enum):
    EMAIL = "EMAIL"
    SMS = "SMS"


class NotificationStatus(enum.Enum):
    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"


class Notification(Base):
    __tablename__ = "notifications"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    notification_type = Column(Enum(NotificationType), nullable=False)
    subject = Column(String, nullable=True)  # Optional - only for EMAIL
    body = Column(String, nullable=False)
    customer_ids = Column(JSON, nullable=False)
    status = Column(Enum(NotificationStatus), nullable=False, default=NotificationStatus.PENDING)
    created_by_user_id = Column(UUID(as_uuid=True), ForeignKey('users.id', ondelete='SET NULL'), nullable=True)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))